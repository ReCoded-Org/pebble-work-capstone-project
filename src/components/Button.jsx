import React from "react";

const Button = ({
    type = "button",
    label = "Get started",
    bgColor = "bg-secondary-300",
    textColor = "text-white",
    fontWeight = "font-normal",
    borderColor,
    hoverBgColor,
    hoverBorderColor,
    height = "h-8",
    width = "w-30",
    onClick,
    border,
    borderWidth,
    fontSize,
}) => {
    return (
        <button
            type={type}
            onClick={(e) => onClick(e)}
            className={`mt-1 mb-1 ${width} ${height} ${fontSize} ${bgColor} ${border} ${borderWidth} hover:${hoverBgColor} ${textColor} ${fontWeight}  py-1  px-2 ${borderColor} hover:${hoverBorderColor} rounded  hover:shadow-lg`}
        >
            {label}
        </button>
    );
};
export default Button;
