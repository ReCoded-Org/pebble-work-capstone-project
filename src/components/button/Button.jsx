import React from "react";

const Button = ({ type = "button", onClick, children }) => {
    //you should put these classes to the div you will use to wrappe the button so you can use it to change buttons styles
    // <div className='w-max rounded ...'>

    return (
        <>
            <button
                type={type}
                onClick={onClick}
                className='rounded border py-2 px-4 text-white'
            >
                {children}
            </button>
        </>
    );
};

export default Button;
