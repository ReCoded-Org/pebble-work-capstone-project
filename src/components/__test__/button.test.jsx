"use strict";

import renderer from "react-test-renderer";

import Button from "../Button";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <Button
                label='Get started'
                bgColor='bg-secondary-300'
                textColor='text-white'
                fontWeight='font-normal'
                height='h-8'
                width='w-30'
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
