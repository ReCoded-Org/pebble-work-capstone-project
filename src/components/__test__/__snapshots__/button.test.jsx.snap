// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="mt-1 mb-1 w-30 h-8 undefined bg-secondary-300 undefined undefined hover:undefined text-white font-normal  py-1  px-2 undefined hover:undefined rounded  hover:shadow-xl"
  onClick={[Function]}
  type="button"
>
  Get started
</button>
`;
